digiroad2-viite/src/test/scala/fi/liikennevirasto/viite/dao/JunctionDAOSpec.scala

```diff
@@ -91,4 +98,70 @@ class JunctionDAOSpec extends FunSuite with Matchers {
+      val roadwayNumber2 = Sequences.nextRoadwayNumber
+      roadwayDAO.create(
+        Seq(
+          Roadway(roadwayDAO.getNextRoadwayId, roadwayNumber1, roadNumber, roadPartNumber, AdministrativeClass.State, Track.Combined, Discontinuity.EndOfRoad, 0, 300, reversed = false, DateTime.parse("1992-10-08"), None, "test", Some("TEST ROAD 1"), 8, TerminationCode.NoTermination),
+          Roadway(roadwayDAO.getNextRoadwayId, roadwayNumber2, roadNumber, roadPartNumber, AdministrativeClass.State, Track.Combined, Discontinuity.EndOfRoad, 300, 500, reversed = false, DateTime.parse("2021-01-01"), None, "test", Some("TEST ROAD 1"), 8, TerminationCode.NoTermination)
+
+        )
+      )
+      // create roadway points
+      val rwpId = roadwayPointDAO.create(
+        RoadwayPoint(Sequences.nextRoadwayPointId, roadwayNumber1, junctionAddrM, "test", Some(DateTime.parse("1992-10-08")), None, None)
+      )
+      val rwpId2 = roadwayPointDAO.create(
+        RoadwayPoint(Sequences.nextRoadwayPointId, roadwayNumber2, endAddrM, "test", Some(DateTime.parse("2021-01-01")), None, None)
+      )
+      // create node points
+      nodePointDAO.create(
+        Seq(
+          NodePoint(Sequences.nextNodePointId, BeforeAfter.Before, rwpId, Some(nodeNumber1), NodePointType.RoadNodePoint, Some(DateTime.parse("1992-10-08")), None, DateTime.parse("1992-10-08"), None, "test", Some(DateTime.parse("1992-10-08")), roadwayNumber1, junctionAddrM, roadNumber, roadPartNumber, Track.Combined, 1, Point(100, 100)),
+          NodePoint(Sequences.nextNodePointId, BeforeAfter.Before, rwpId2, Some(nodeNumber2), NodePointType.RoadNodePoint, Some(DateTime.parse("2021-01-01")), None, DateTime.parse("2021-01-01"), None, "test", Some(DateTime.parse("2021-01-01")), roadwayNumber2, endAddrM, roadNumber, roadPartNumber, Track.Combined, 1, Point(100, 200))
+        )
+      )
+      // create junction
+      val junctionId = Sequences.nextJunctionId
+      dao.create(
+        Seq(
+          Junction(junctionId, Some(1), Some(nodeNumber1), DateTime.parse("1992-10-08"), None, DateTime.parse("1992-10-08"), None, "test", Some(DateTime.parse("1992-10-08")), None)
+        )
+      )
+      // create junction points
+      junctionPointDAO.create(
+        Seq(
+          JunctionPoint(Sequences.nextJunctionPointId, BeforeAfter.Before, rwpId, junctionId, Some(DateTime.parse("1992-10-08")), None, DateTime.parse("1992-10-08"), None, "test", Some(DateTime.parse("1992-10-08")), roadwayNumber1, junctionAddrM, roadNumber, roadPartNumber, Track.Combined, Discontinuity.Continuous, Point(100,100)),
+          JunctionPoint(Sequences.nextJunctionPointId, BeforeAfter.After, rwpId, junctionId, Some(DateTime.parse("1992-10-08")), None, DateTime.parse("1992-10-08"), None, "test", Some(DateTime.parse("1992-10-08")), roadwayNumber1, junctionAddrM, roadNumber, roadPartNumber, Track.Combined, Discontinuity.Continuous, Point(100,100))
+        )
+      )
+
+      // fetch and test
```

Comment:
Tykkäisin testattavien selitteistämisestä, jotain niinku:

// fetch and test 
// junctions (of the road _roadnumber_, created(?) before 2022-01-01 (there should be 1, with e&j topology markers)" 
...
// junctions (of the road _roadnumber_, created(?) before 1992-01-01 (there should be 0)" 

tai jotain
(...oliks toi nyt luontipäivä, johon tota päiväysparametria verrattiin...)